pkg/controller/deployment/deployment_controller.go

```diff
@@ -424,6 +424,27 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+// syncPausedDeployment allows scaling of the deployment (either done manually
+// or by autoscalers) to propagate to the active rc of the deployment.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, err := deploymentutil.GetNewRCFromList(deployment, dc.client,
+		func(namespace string, options api.ListOptions) ([]api.ReplicationController, error) {
+			return dc.rcStore.List()
+		})
+	if err != nil {
+		return err
+	}
+	if newRC == nil {
+		// There is no replication controller that matches the intent of
+		// this deployment which means that either the deployment has never
+		// run (created as paused), or it was switched to paused and has been
+		// modified since. Such is life.
+		return nil
+	}
+	_, err = dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
```

Comment:
Why might a user pause a rollout? They might want to treat new instances as a canary to observe how they behave under real load. Or they might want to run tests in prod. Or they may have observed a problem that they want to investigate.

In any of these cases, I'd imagine they'd want to freeze the rollout in whatever state it was in.

And, whether paused or not, I'd imagine scaling (eg. by HPA) would change the current state proportionally.